core/src/main/scala/io/finch/endpoint/Paths.scala

```diff
@@ -0,0 +1,87 @@
+package io.finch.endpoint
+
+import cats.effect.Effect
+import io.finch._
+import io.netty.handler.codec.http.QueryStringDecoder
+import scala.reflect.ClassTag
+import shapeless.HNil
+
+private[finch] class Paths[F[_] : Effect] {
```

Comment:
I'm not yet sold on this new way of providing endpoint instances. I think there are good arguments could be made for both approaches.

```
// iteratee-style (what's currently implemented)
import io.finch.arrows._
val b: Endpoint[Task, Foo] = jsonBody[Foo]

// fs2 (and others) style
val b = jsonBody[Task, Foo]
```

I feel like adopting the former somewhat defeats the purpose of `cats-effect` as we'd need to take dependencies on different effect implementations to provide people with `io.finch.yourFancyEffect` With just `jsonBody[Task, Foo]` we would not need to do that.

An alternative (and a middle ground) solution would be to allow people create their own modules given the `Effect` instance.

```
val f = io.Finch[Task]
import f._
val b = jsonBody[Foo]
```

I want to spend some time to look around and see what other libraries are doing before I can lean towards any of these approaches.

Reply:
There is no need for us to provide `io.finch.yourFancyEffect` as people might implement their own `object yourFancyEffect` using same approach as https://github.com/finagle/finch/pull/979/files#diff-5c1246ec310c1d425749036da9b6bb61
A lot of traits would be simplified to something like `abstract class Finch[F[_] : Effect]`

And they will get all the endpoints for free

Reply:
@vkostyukov check out `abstract class Finch` in the latest commit plox